tests/zenroom_proposal.py

```diff
@@ -0,0 +1,206 @@
+Given I have a 'signature' named 'data.signature' inside 'result'
+When I verify the 'houses' has a signature in 'data.signature' by 'Alice'
+Then print the string 'ok' 
+"""
+# CRYPTO-CONDITIONS: instantiate an Ed25519 crypto-condition for buyer
+zenSha = ZenroomSha256(script=fulfill_script, keys=zen_public_keys, data=data)
+
+# CRYPTO-CONDITIONS: generate the condition uri
+condition_uri = zenSha.condition.serialize_uri()
+# CRYPTO-CONDITIONS: construct an unsigned fulfillment dictionary
+unsigned_fulfillment_dict = {
+    'type': zenSha.TYPE_NAME,
+    'script': fulfill_script,
+    'keys': zen_public_keys,
+}
+
+output = {
+    'amount': '1000',
+    'condition': {
+        'details': unsigned_fulfillment_dict,
+        'uri': condition_uri,
+    },
+    'data': data,
+    'script': fulfill_script,
+    'conf': '',
+    'public_keys': (zen_public_keys['Alice']['ecdh_public_key'], ),
+}
+
+
+input_ = {
+    'fulfillment': None,
+    'fulfills': None,
+    'owners_before': (zen_public_keys['Alice']['ecdh_public_key'], ),
+}
+
+token_creation_tx = {
+    'operation': 'CREATE',
+    'asset': asset,
+    'metadata': None,
+    'outputs': (output,),
+    'inputs': (input_,),
+    'version': version,
+    'id': None,
+}
+
+# JSON: serialize the transaction-without-id to a json formatted string
+message = json.dumps(
+    token_creation_tx,
+    sort_keys=True,
+    separators=(',', ':'),
+    ensure_ascii=False,
+)
+
+print("====== GENERATE RESULT (METADATA) =======")
+condition_script = """Rule input encoding base58
+Rule output encoding base58
+Scenario 'ecdh': create the signature of an object 
+Given I have the 'keys'
+Given that I have a 'string dictionary' named 'houses' inside 'asset' 
+When I create the signature of 'houses' 
+When I rename the 'signature' to 'data.signature' 
+Then print the 'data.signature'
+"""
+
+# THIS FILLS THE METADATA WITH THE RESULT
+assert(not zenSha.validate(message=message))
+message = zenSha.sign(message, condition_script, alice)
+assert(zenSha.validate(message=message))
+# now metadata looks like
+# 'metadata': {'result': {'data.signature': {'r': 'fdoan0GYo9RGP8y0fq+PKZ9Q1V8+VqJtBkSMB1tUnGQ=', 's': 'RnJCEepYJcVgFG/Y6cRc/2DWPaz5Pe5NpdRWegrZk5A='}}}
+
+# CRYPTO-CONDITIONS: generate the fulfillment uri
+fulfillment_uri = zenSha.serialize_uri()
+
+# add the fulfillment uri (signature)
+token_creation_tx['inputs'][0]['fulfillment'] = fulfillment_uri
+print(token_creation_tx)
+
+# JSON: serialize the id-less transaction to a json formatted string
+json_str_tx = json.dumps(
+    token_creation_tx,
+    sort_keys=True,
+    separators=(',', ':'),
+    ensure_ascii=False,
+)
+
+
+# SHA3: hash the serialized id-less transaction to generate the id
+shared_creation_txid = hashlib.sha3_256(json_str_tx.encode()).hexdigest()
+
+# add the id
+token_creation_tx['id'] = shared_creation_txid
+
+exit()
+# send CREATE tx into the bdb network
+returned_creation_tx = bdb.transactions.send_async(token_creation_tx)
```

Comment:
BDB is not used here, the tests are also not recognized by the test-framework. please have a look at how the other tests are marked, and add zenroom_proposal.py to the test-suite so that it is executed and verified by running make test or tox. feel free to ask questions. I am happy to support you

Reply:
That file didn't want to be a unit test, it just want to be an example of usage of the zenroom cryptoconditions. It doesn't use BDB because the zenroom cryptocondition has to be included in the code of bigchaindb/planetmint and I haven't done it.
What do you think about leaving that file as it is, maybe we can move it somewhere else, because at the moment it is the file that shows how to use the zenroom cryptoconditions

Reply:
At the moment, with your help, I can write a test for the validation, but I would argue that it is early, I am not even sure that the structure of the cryptocondition will remain the same...